lib/game.rb

```diff
@@ -59,48 +66,52 @@ def board_setup
     p "You have successfully placed your ships. PREPARE TO DIE!"
   end
 
-  def turn
+  def user_turn
     puts "=============COMPUTER BOARD============="
     # take out render true!
     print @cpu_board.render
     puts "==============PLAYER BOARD=============="
     print @p1_board.render(true)
     puts "Enter the coordinate for your shot with the letter followed by the number (i.e. D4):"
     user_input = gets.chomp
-      if @cpu_board.valid_coordinate?(user_input)
-      @cpu_board.cells[user_input].fire_upon
-        if @cpu_board.cells[user_input].empty?
-          p1_hit_or_miss = "a miss."
-        else
-          p1_hit_or_miss = "a hit."
-        end
-      cpu_choice = @p1_board.cells.keys.sample
-      until @p1_board.cells[cpu_choice].fired_upon? == false
-        cpu_choice = @p1_board.cells.keys.sample
+    if @cpu_board.valid_coordinate?(user_input) && 
+    @cpu_board.cells[user_input].fire_upon
+      if @cpu_board.cells[user_input].empty?
+        p1_hit_or_miss = "a miss."
+      else
+        p1_hit_or_miss = "a hit."
       end
-      @p1_board.cells[cpu_choice].fire_upon
-        if @p1_board.cells[cpu_choice].empty?
-          cpu_hit_or_miss = "a miss."
-        else
-          cpu_hit_or_miss = "a hit."
-        end
       puts "Your shot on #{user_input} was #{p1_hit_or_miss}"
-      puts "My shot on #{cpu_choice} was #{cpu_hit_or_miss}"
       @p1_board.cells
-      else
-      puts "Invalid coordinate. Please try again."
-      self.turn
-      end
+    else
+    puts "Invalid coordinate. Please try again."
+    end
+  end
+  
+  def game_over?
     if @cpu_cruiser.sunk? && @cpu_submarine.sunk? 
       puts "Impossible, how did you do this? HOW could you? I have a fam..."
       start
     elsif @p1_cruiser.sunk? && @p1_submarine.sunk?
       puts "Everything as it should be. All your ship are belong to us!"
       start
-    else self.turn
     end
   end
 
+  def cpu_turn
```

Comment:
Nice! I like that you've made this a separate method. Turn was a monster beforehand.